### PR TITLE
Adding `detail`s element to Core -> Base

### PR DESCRIPTION
**Pull request by chris-pearce**

### CODE REVIEW

**Comment by AlecRaeside** [comment]
LGTM

File: core/base/_details.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  /**
+   * Control the visibility of the `detail`s marker.
+   */
+
+  @if $details-marker {
+
+    &::-webkit-details-marker {display: none;}
+
+  }
```
Comment (kllevin) [inline core/base/_details.scss:31]: Is this saying that if I had set `details-marker` to true it would still be hidden?

Comment (chris-pearce) [inline core/base/_details.scss:31]: The default is not hidden, setting it too true will hide it.

Comment (kllevin) [inline core/base/_details.scss:31]: :+1:

**Comment by kllevin** [comment]
LGTM

**Pull request merged**